lib/llm/src/discovery/watcher.rs

```diff
@@ -151,13 +173,91 @@ impl ModelWatcher {
             .await
             .with_context(|| model_name.clone())?;
         if !active_instances.is_empty() {
+            let mut update_tx = true;
+            let mut model_type: ModelType = model_entry.model_type;
+            if model_entry.model_type == ModelType::Chat
+                && self.manager.list_chat_completions_models().is_empty()
+            {
+                self.manager.remove_chat_completions_model(&model_name).ok();
```

Comment:
@abrarshivani sorry for commenting on old PR, can you help me understand the conditional here? `remove_chat_completions_model` will be run if `list_chat_completions_models().is_empty()`, but in that case, wouldn't the `remove` do nothing as the manger already has no chat completions models?